### PR TITLE
SecurtyContext: applying PodTemplate for InitContainers and removing SYS_PTRACE 

### PR DESCRIPTION
**Pull request by vlibov**
## Tracking issue

Related to #5462

## Why are the changes needed?

The Issue https://github.com/flyteorg/flyte/issues/5462 describes the problem in detail.

## What changes were proposed in this pull request?

1. Remove the SYS_PTRACE capability
2. Apply the PodTemplate settings to the InitContainers. In particular, this allows the SecurityContext settings to be applied also to the InitContainer.

## Check all the applicable boxes 

- [ ] I updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] All commits are signed-off.

### CODE REVIEW

File: flyteplugins/go/tasks/pluginmachinery/flytek8s/copilot.go [hunk 1]
```diff
@@ -172,7 +172,10 @@ func AddCoPilotToContainer(ctx context.Context, cfg config.FlyteCoPilotConfig, c
 	if c.SecurityContext.Capabilities == nil {
 		c.SecurityContext.Capabilities = &v1.Capabilities{}
 	}
-	c.SecurityContext.Capabilities.Add = append(c.SecurityContext.Capabilities.Add, pTraceCapability)
+	var add_sys_ptrace bool = false
+	if add_sys_ptrace {
```
Comment (pingsutw) [inline flyteplugins/go/tasks/pluginmachinery/flytek8s/copilot.go:176]: Isn't it always false here? Should we add `add_sys_ptrace` to the plugin config?

Comment (hamersaw) [reply flyteplugins/go/tasks/pluginmachinery/flytek8s/copilot.go:176]: If there is a larger conversation here, which I suspect there is. We can split this into multiple PRs - I don't think merging support for applying a PodTemplate to initContainers is controversial.

Comment (davidmirror-ops) [reply flyteplugins/go/tasks/pluginmachinery/flytek8s/copilot.go:176]: >We can split this into multiple PRs
Agree. @vlibov thoughts?

Comment (vlibov) [reply flyteplugins/go/tasks/pluginmachinery/flytek8s/copilot.go:176]: Hi @davidmirror-ops,
how do we want to proceed? Shall we split the PR or make the SYS_PTRACE configurable (or remove it completely?)
Thanks!

Comment (davidmirror-ops) [reply flyteplugins/go/tasks/pluginmachinery/flytek8s/copilot.go:176]: >Shall we split the PR or make the SYS_PTRACE configurable
Hey @vlibov! Can we do both? Split this into two PRs: one for the PodTemplate for initContainer and the other making SYS_PTRACE configurable.

Comment (vlibov) [reply flyteplugins/go/tasks/pluginmachinery/flytek8s/copilot.go]: Sounds good, @davidmirror-ops!
I reverted the SYS_PTRACE commit in this PR so it now only affects the initContainer part.

I will also create a new PR for the SYS_PTRACE config - can anybody point me on how to properly implement this? i.e. where is this option supposed to be defined and how to propagate it down to the relevant function which sets SYS_PTRACE? Thanks!

Comment (vlibov) [reply flyteplugins/go/tasks/pluginmachinery/flytek8s/copilot.go]: Hey @davidmirror-ops, actually I created a new PR also for this change (initContainers)
https://github.com/flyteorg/flyte/pull/5664
I hope this is ok!
The #5556 can thus be closed...

Comment (dylanspag-lmco) [reply flyteplugins/go/tasks/pluginmachinery/flytek8s/copilot.go]: Hi @vlibov, just wondering did you end up creating a separate PR for removing SYS_PTRACE?

**Pull request closed without merging**